Backend/src/main/java/com/onbank/exceptions/advice/ExceptionControllerAdvice.java

```diff
@@ -0,0 +1,30 @@
+package com.onbank.exceptions.advice;
+
+import com.onbank.TestEndPoint;
+import com.onbank.exceptions.ExceptionResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+
+@ControllerAdvice
+public class ExceptionControllerAdvice {
+
+    //Nieznany blad
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity <ExceptionResponse> unknownException(final Exception e) {
+        ExceptionResponse exceptionResponse = new ExceptionResponse();
+        exceptionResponse.setCode(HttpStatus.INTERNAL_SERVER_ERROR.value());
+        exceptionResponse.setDescription("Exception msg: " + e.toString());
+        return new ResponseEntity<>(exceptionResponse, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(NullPointerException.class)
+    public ResponseEntity <ExceptionResponse> nullPointerException(final Exception e) {
+        ExceptionResponse exceptionResponse = new ExceptionResponse();
+        exceptionResponse.setCode(HttpStatus.INTERNAL_SERVER_ERROR.value());
+        exceptionResponse.setDescription("Null Pointer Exception");
```

Comment:
Skoro w handlerze wyżej jest wrzucenie do Description opisu wyjątku to nie ma sensu dodatkowe obsługiwanie NPE

Reply:
Napisałem dodatkowy NPE ponieważ taki był wymóg zapisany na trello, miała być obsługa nieznanych wyjątków i takich zaimplementowanych. + w przypadku znanych może zmienić się HttpStatus. (Tutaj wygląd i działanie metod jest zbliżony, gdyż mają bardziej pokazywać jak w przyszłości implementować obsługe wyjątków.)

Reply:
NPE to nie jest znany wyjątek, to jest nieprzewidziany błąd występujący w trakcie działania aplikacji. Zaimplementowany wyjątek to wyjątek biznesowy, o którego wyrzuceniu decyduje programista piszący aplikacjie, np wyjątek ItemNotFoundException który wyrzucamy w przypadku gdy nie znajdziemy wymaganego przez klienta elementu